sale_partner_insurance/models/sale_order.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2024 Akretion France (http://www.akretion.com/)
+# @author: Mathieu Delva <mathieu.delva@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def action_confirm(self):
```

Comment:
after confirmation, we can change some values which impact the sale price. We need to recalculate the insurance line price_unit in that case. (in a write method for example)